docs/getting_started/typescript/hello_world_in_typescript/index.md

```diff
@@ -17,67 +18,136 @@ image: /img/temporal-logo-twitter-card.png
+Creating reliable applications is a difficult chore. [Temporal](https://temporal.io/) lets you create fault-tolerant, resilient applications using programming languages you already know, so you can build complex applications that execute reliably and recover from failures.
 
-In this tutorial, you will build your first Temporal Application from scratch using the [Temporal TypeScript SDK](https://github.com/temporalio/sdk-typescript). The app will consist of four pieces:
+In this tutorial, you will build your first Temporal Application from scratch using the [Temporal TypeScript SDK](https://github.com/temporalio/sdk-typescript). You'll develop a small application that asks for your name and then uses APIs to get your public IP address and your location based on that address. External requests can fail due to rate limiting, network interruptions, or other errors. Using Temporal for this application will let you automatically recover from these and other kinds of failures without having to write explicit error-handling code.
 
-1. A [Workflow](https://docs.temporal.io/workflows): Workflows are functions that define the overall flow of the application.
-2. An [Activity](https://docs.temporal.io/activities): Activities are functions called by Workflows, and they contain any logic that might fail or behave differently at different times. The Workflow you'll create executes a single Activity that takes a string as input and returns a formatted version of this string to the Workflow.
-3. A [Worker](https://docs.temporal.io/workers): Workers host the Activity and Workflow code and executes the code piece by piece.
-4. An initiator: To start a Workflow, you need to create a Client instance and call the `.start()` method. You'll write a separate script to do this.
+The app will consist of the following pieces:
 
-You'll also write a unit test to ensure your Workflow executes successfully.
+1. Two [Activities](https://docs.temporal.io/activities): Activities are functions called by Workflows, and they contain any logic that might fail or behave differently at different times. The first Activity will get your IP address, and the second Activity will use that IP address to get your location.
 
-When you're done, you'll have a basic application and a clear understanding of how to build out the components you'll need in future Temporal applications.
+1. A [Workflow](https://docs.temporal.io/workflows): Workflows are functions that define the overall flow of the application.  Your Workflow will execute both Activities you define. It will call the first Activity to fetch the IP address, and then use the result of that Activity to call the next Activity to find your location.
+2. A [Worker](https://docs.temporal.io/workers): Workers host the Activity and Workflow code and executes the code piece by piece.
+3. A client program: Youll need to start your Worfklow. In this tutorial you'll create a small program using the Temporal Client to start the Workflow.
```

Comment:
Do you want to link to a page on the Client since you linked Activity, Workflow, and Worker? https://docs.temporal.io/encyclopedia/temporal-sdks#temporal-client

Reply:
```suggestion
3. A client program: Youll need to start your Worfklow. In this tutorial you'll create a small program using the [Temporal Client](https://docs.temporal.io/encyclopedia/temporal-sdks#temporal-client) to start the Workflow.
```